src/datatypes/src/vectors/binary.rs

```diff
@@ -36,6 +36,40 @@ impl BinaryVector {
     pub(crate) fn as_arrow(&self) -> &dyn Array {
         &self.array
     }
+
+    /// Creates a new binary vector of JSONB from a binary vector.
+    /// The binary vector must contain valid JSON strings.
+    pub fn convert_binary_to_json(&self) -> Result<BinaryVector> {
+        let arrow_array = self.to_arrow_array();
+        let mut vector = vec![];
+        for binary in arrow_array
+            .as_any()
+            .downcast_ref::<BinaryArray>()
+            .unwrap()
+            .iter()
+        {
+            let jsonb = if let Some(binary) = binary {
+                match jsonb::parse_value(binary) {
+                    Ok(jsonb) => Some(jsonb.to_vec()),
+                    Err(_) => {
+                        if jsonb::parse_jsonb(binary).is_ok() {
```

Comment:
Can we prioritize assuming the binary is in jsonb format? Jsonb::parse_jsonb only needs to parse the first 4 bytes of the header.

Reply:
Got you. We can use `from_slice` here which first trys to parse as jsonb and then parse as json strings if failed.